docs/graph-algorithms/breadth-first-search.mdx

```diff
@@ -0,0 +1,416 @@
+---
+title: Breadth First Search
+slug: /graph-algorithms/breadth-first-search
+---
+
+## Introduction
+Breadth-first search (BFS) is an algorithm for traversing or searching tree or graph data structures.
+It starts at the tree root (or some arbitrary node of a graph, sometimes referred to as a 'search key'[1]),
+and explores all of the neighbor nodes at the present depth prior to moving on to the nodes at the next depth level.
+
+## Explanation
+
+- Given an undirected & Connected Graph (V, E) <br />
+  V: Vertices E: Edges <br />
+- Goal: methodically explore every vertex and every edge <br />
+- “Explore” a graph, turning it into a tree<br />
+- One vertex at a time<br />
+- Expand frontier of explored vertices across its breadth<br />
+- Discovers all vertices at distance k from s before those at distance k+1
+- A level-by-level order traversal<br />
+- For example, in the following graph, we start traversal from vertex 2. When we come to vertex 0, we look for all adjacent vertices of it. 2 is also an adjacent vertex of 0. If we don’t mark visited vertices, then 2 will be processed again and it will become a non-terminating process.
+ A Breadth First Traversal of the following graph is 2, 0, 3, 1.<br />
+ [![graph.png](https://media.geeksforgeeks.org/wp-content/uploads/bfs-5.png)]<br />
+
+## Applications of Breadth First Traversal
+
+- Find Connected Components in Undirected G <br />
+- Web Crawling<br />
+- Social N/W: Friend Finder<br />
+- Pocket Cube (2x2x2 Rubic Cube)<br />
+- Garbage Collector<br />
+
+
+## Algorithm
+
+- Builds a tree over the graph
+- Pick a source vertex to be the root
+- Find (“discover”) all of its children, then all of their children, etc.
+- Associate vertex “colors” to guide the algorithm
+- white  vertices have not been discovered
+- All vertices start out white
+- Grey vertices are discovered but not fully explored
+- They may be adjacent to white vertices
+- Black vertices are discovered and fully explored
+- They are adjacent only to black and gray vertices
+- Explore vertices by scanning neighbors of grey vertices
+
+## Code
```

Comment:
Please refer to other merged PRs to see how to write code in different languages.
You need to add something like 
```
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

<Tabs
  defaultValue="cpp"
  values={[
    { label: 'C++', value: 'cpp', },
    { label: 'C', value: 'c', },
    { label: 'Java', value: 'java', }
  ]
}>
```